pkg/controller/expcertificates/util.go

```diff
@@ -203,3 +203,52 @@ func SecretDataAltNamesMatchSpec(secret *corev1.Secret, spec cmapi.CertificateSp
 
 	return violations, nil
 }
+
+// staticTemporarySerialNumber is a fixed serial number we use for temporary certificates
+const staticTemporarySerialNumber = "1234567890"
+
+// GenerateLocallySignedTemporaryCertificate signs a temporary certificate for
+// the given certificate resource using a one-use temporary CA that is then
+// discarded afterwards.
+// This is to mitigate a potential attack against x509 certificates that use a
+// predictable serial number and weak MD5 hashing algorithms.
+// In practice, this shouldn't really be a concern anyway.
+func GenerateLocallySignedTemporaryCertificate(crt *cmapi.Certificate, pkData []byte) ([]byte, error) {
```

Comment:
Does this need to be exported? Seems like it is only referenced from one package and the exact definition of this function _could_ change in future... feel like we may be best to not export and move this into `issuing/`?

Reply:
This is used in two places
```
./internal/test/test.go:177:	tempCertBytes, err := certificates.GenerateLocallySignedTemporaryCertificate(crt, privateKeyBytes)
./issuing/issuing_controller.go:139:		localTemporarySigner:     certificates.GenerateLocallySignedTemporaryCertificate,
./util.go:210:// GenerateLocallySignedTemporaryCertificate signs a temporary certificate for
./util.go:216:func GenerateLocallySignedTemporaryCertificate(crt *cmapi.Certificate, pkData []byte) ([]byte, error) {
```

Reply:
🤦 my bad :)